src/SMatrix.jl

```diff
@@ -29,10 +29,6 @@ immutable SMatrix{S1, S2, T, L} <: StaticMatrix{T}
 end
 
 @generated function check_smatrix_params{S1,S2,L}(::Type{Val{S1}}, ::Type{Val{S2}}, T, ::Type{Val{L}})
-    if !(T <: DataType) # I think the way types are handled in generated fnctions might have changed in 0.5?
```

Comment:
Actually looks like the correct thing to put there is `if !(T isa Type{<:Type})` since we're in a generated function. There's no way to do this in 0.5 AFAIK, although you can always just check outside the generated function. This solution of just deleting these lines at least gives the same error messages between SMatrix/Array/Vector when you try to construct them with a non-type T, albeit crytic. I could look more tomorrow into catching this error more directly if you'd like.